bin/s2nc.c

```diff
@@ -523,7 +549,12 @@ int main(int argc, char *const *argv)
             GUARD_EXIT(s2n_connection_set_session(conn, session_state, session_state_length), "Error setting session state in connection");
         }
 
-        GUARD_EXIT(s2n_setup_external_psk_list(conn, psk_optarg_list, psk_list_len), "Error setting external psk list"); 
+        GUARD_EXIT(s2n_setup_external_psk_list(conn, psk_optarg_list, psk_list_len), "Error setting external psk list");
+
+        if (early_data && session_state_length) {
```

Comment:
Should this just be "if(early_data)" and we exit if session resumption isn't enabled? We can't do early data without session resumption, after all.

Reply:
We want to send early data only if we currently have a session ticket. Otherwise we'd be trying to send early data on the first full handshake, which is incorrect.

Reply:
Oh yeah for sure. I'm just suggesting an error instead of skipping early data.